### PR TITLE
8328227: Remove applet usage from JColorChooser tests Test4887836

### PR DESCRIPTION
**Pull request by alisenchung**
Removing applet usage from manual JColorChooser tests

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8328227](https://bugs.openjdk.org/browse/JDK-8328227): Remove applet usage from JColorChooser tests Test4887836 (**Bug** - P4)

### Reviewers
 * [Harshitha Onkar](https://openjdk.org/census#honkar) (@honkar-jdk - **Reviewer**)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to [6ff8dfa4](https://git.openjdk.org/jdk/pull/18316/files/6ff8dfa40f21139ac656d334a6158c88901a142d)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18316/head:pull/18316` \
`$ git checkout pull/18316`

Update a local copy of the PR: \
`$ git checkout pull/18316` \
`$ git pull https://git.openjdk.org/jdk.git pull/18316/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18316`

View PR using the GUI difftool: \
`$ git pr show -t 18316`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18316.diff">https://git.openjdk.org/jdk/pull/18316.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18316#issuecomment-1998624933)

### CODE REVIEW

File: test/jdk/javax/swing/JColorChooser/Test4887836.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2003, 2008, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2003, 2008, 2024 Oracle and/or its affiliates. All rights reserved.
```
Comment (honkar-jdk) [inline test/jdk/javax/swing/JColorChooser/Test4887836.java:2]: Same here. Please remove 2008 and add comma.

File: test/jdk/javax/swing/JColorChooser/Test4887836.java [hunk 2]
```diff
@@ -21,23 +21,40 @@
...
  * @summary Checks if no tooltip modification when no KeyStroke modifier
- * @author Konstantin Eremin
- * @run applet/manual=yesno Test4887836.html
+ * @run main/manual Test4887836
  */
+public class Test4887836 {
+
+    public static void main(String[] args) throws Exception {
+        String instructions = "If you do not see white area under the \"Swatches\" tab,\n" +
+                "then test passed, otherwise it failed.";
```
Comment (honkar-jdk) [inline test/jdk/javax/swing/JColorChooser/Test4887836.java:41]: The original instructions seem a bit vague and the test window has a lot going on so probably more details here would add clarity.

File: test/jdk/javax/swing/JColorChooser/Test4887836.java [hunk 3]
```diff
@@ -21,23 +21,40 @@
...
+
+        PassFailJFrame.builder()
+                .title("Test4759306")
+                .instructions(instructions)
+                .rows(5)
+                .columns(40)
+                .testTimeOut(10)
+                .splitUIRight(Test4887836::createColorChooser)
```
Comment (honkar-jdk) [inline test/jdk/javax/swing/JColorChooser/Test4887836.java:49]: Here having separate windows for instructions and test window look better.
```suggestion
                .testUI(Test4887836::createColorChooser)
```

File: test/jdk/javax/swing/JColorChooser/Test4887836.java [hunk 4]
```diff
@@ -21,23 +21,40 @@
...
+import java.awt.Font;
+import javax.swing.JColorChooser;
+import javax.swing.UIManager;
+
 /*
  * @test
  * @bug 4887836
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
  * @summary Checks if no tooltip modification when no KeyStroke modifier
```
Comment (honkar-jdk) [inline test/jdk/javax/swing/JColorChooser/Test4887836.java:34]: @alisenchung summary doesn't match with the test.

**Review by honkar-jdk — CHANGES_REQUESTED** [review]

File: test/jdk/javax/swing/JColorChooser/Test4887836.java [hunk 5]
```diff
@@ -21,23 +21,40 @@
...
- * @author Konstantin Eremin
- * @run applet/manual=yesno Test4887836.html
+ * @run main/manual Test4887836
  */
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/JColorChooser/Test4887836.java:36]: May add a blank line after this for better readability.

File: test/jdk/javax/swing/JColorChooser/Test4887836.java [hunk 6]
```diff
@@ -21,23 +21,40 @@
...
+                .splitUIRight(Test4887836::createColorChooser)
+                .build()
+                .awaitAndCheck();
+    }
 
-import java.awt.Color;
-import java.awt.Font;
-import javax.swing.JApplet;
-import javax.swing.JColorChooser;
-import javax.swing.UIManager;
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/JColorChooser/Test4887836.java:54]: can be removed.

File: test/jdk/javax/swing/JColorChooser/Test4887836.java [hunk 7]
```diff
@@ -21,23 +21,40 @@
...
 
-public class Test4887836 extends JApplet {
-    public void init() {
+    private static JColorChooser createColorChooser() {
+        JColorChooser chooser = new JColorChooser(Color.LIGHT_GRAY);
         UIManager.put("Label.font", new Font("Perpetua", 0, 36)); // NON-NLS: property and font names
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/JColorChooser/Test4887836.java:57]: Can add comment above.
```suggestion
        // NON-NLS: property and font names
        UIManager.put("Label.font", new Font("Perpetua", 0, 36));
```

File: test/jdk/javax/swing/JColorChooser/Test4887836.java (hunk 2 above)
Comment (alisenchung) [reply test/jdk/javax/swing/JColorChooser/Test4887836.java:41]: The original test [JDK-4887836](https://bugs.openjdk.org/browse/JDK-4887836) is quite vague so I'm not sure how to add more details other than "white area under 'Swatches' tab"

File: test/jdk/javax/swing/JColorChooser/Test4887836.java [hunk 8]
```diff
@@ -21,23 +21,42 @@
...
 
-import java.awt.Color;
-import java.awt.Font;
-import javax.swing.JApplet;
-import javax.swing.JColorChooser;
-import javax.swing.UIManager;
+public class Test4887836 {
+
+    public static void main(String[] args) throws Exception {
+        String instructions = "If you do not see white area under the \"Swatches\" tab,\n" +
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/JColorChooser/Test4887836.java:41]: You may use `Text Block` for clean instructions.

> """
> Your Instructions. """;

File: test/jdk/javax/swing/JColorChooser/Test4887836.java [hunk 9]
```diff
@@ -21,23 +21,43 @@
...
+
+    private static JColorChooser createColorChooser() {
+        JColorChooser chooser = new JColorChooser(Color.LIGHT_GRAY);
 
-public class Test4887836 extends JApplet {
-    public void init() {
-        UIManager.put("Label.font", new Font("Perpetua", 0, 36)); // NON-NLS: property and font names
-        add(new JColorChooser(Color.LIGHT_GRAY));
+        // NON-NLS: property and font names
+        UIManager.put("Label.font", new Font("Perpetua", 0, 36));
```
Comment (prrace) [inline test/jdk/javax/swing/JColorChooser/Test4887836.java:60]: Sorry, not related to the applet conversion but Perpetua ?? Pointless.  Won't exist.
Change it to Font.DIALOG. I think you should also remove the comment line.

**Review by prrace — APPROVED** [review]

File: test/jdk/javax/swing/JColorChooser/Test4887836.java (hunk 2 above)
Comment (honkar-jdk) [reply test/jdk/javax/swing/JColorChooser/Test4887836.java]: Which "White area" are they referring in Swatches tab? Is it the preview window?

File: test/jdk/javax/swing/JColorChooser/Test4887836.java [hunk 10]
```diff
@@ -21,23 +21,42 @@
...
-import java.awt.Font;
-import javax.swing.JApplet;
-import javax.swing.JColorChooser;
-import javax.swing.UIManager;
+public class Test4887836 {
+
+    public static void main(String[] args) throws Exception {
+        String instructions = """
+                                If you do not see white area under the \"Swatches\" tab,\n +
+                                then test passed, otherwise it failed.""";
```
Comment (honkar-jdk) [inline test/jdk/javax/swing/JColorChooser/Test4887836.java:43]: please fix line-lengths and the "+" operator is not required since String blocks are used.

Comment (honkar-jdk) [reply test/jdk/javax/swing/JColorChooser/Test4887836.java]: Please fix line-lengths.

**Review by honkar-jdk — APPROVED** [review]

**Pull request closed without merging**